app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -154,6 +156,25 @@ class BrowserWebViewClient(
                     }
                     false
                 }
+                is SpecialUrlDetector.UrlType.ExtractedTrackingLink -> {
+                    if (isForMainFrame) {
+                        webViewClientListener?.startProcessingTrackingLink()
```

Comment:
Did you find a scenario where we need to call this in here?

Reply:
This is only needed to update the progress bar correctly (To prevent it loading twice), it does not affect the actual extraction logic.

Reply:
thanks! it makes sense now 👍